pkg/api/validation/validation.go

```diff
@@ -1860,7 +1860,11 @@ func ValidateVolumeMounts(mounts []api.VolumeMount, volumes sets.String, fldPath
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("mountPath"), mnt.MountPath, "must be unique"))
 		}
 		if !path.IsAbs(mnt.MountPath) {
-			allErrs = append(allErrs, field.Invalid(idxPath.Child("mountPath"), mnt.MountPath, "must be an absolute path"))
+			// also allow windows absolute path
+			p := mnt.MountPath
+			if len(p) < 2 || ((p[0] < 'A' || p[0] > 'Z') && (p[0] < 'a' || p[0] > 'z')) || p[1] != ':' {
```

Comment:
This now allows linux users to specify a non-absolute path.  What happens when a linux user says `"C:"` for this field?

And where are the tests for this?  No test == no merge.

Come on folks.  We need to do better than this.  I don't have bandwidth to review every PR, but I shouldn't have to.  Every time I randomly sample and I find things like this, I fear for the project.  You HAVE TO consider the wider implications of EVERY change.  We're well past the days where such changes could be made cavalierly.

I don't know what the solution is, but I need a solution and a test ASAP or I will roll this back. 

Maybe we can redefine this field to always be at the root directory if the path is not absolute, and just remove this validation entirely?  What bad things would happen?

@saad-ali @kubernetes/sig-storage-api-reviews

Reply:
The correct approach is to use the `path/filepath` `isAbs` function:

https://golang.org/pkg/path/filepath/#IsAbs

See:

https://github.com/golang/go/issues/1483

For details.

Reply:
PR with the fix is here:

https://github.com/kubernetes/kubernetes/pull/54995

Reply:
This presumes that the server on which the validation runs is the same OS as the server on which the pod will run.  That is certainly not guaranteed.  will comment there.

Reply:
Apologies if this came off harshly.  What I hoped to convey was that a) we're in a place now where changes such as this have subtle and non-local implications that we HAVE to work through; and b) PRs without tests need to be the rare exception case.  We owe it to our users and each other to raise the quality level on every PR.

Reply:
Hi @thockin, sorry I should add tests for this PR, here is the new test PR: https://github.com/kubernetes/kubernetes/pull/55137

The main difficulty for this fix is:  **validation would not only run on agent node, it also runs on master node, while from the master, it did not know whether the mount path is for linux or agent node.**  (For windows based k8s cluster, the master node is a linux server and agent is windows server)

The absolute path validation was introduced from `v1.8` and it did not consider the condition for windows node, see: https://github.com/kubernetes/kubernetes/commit/8d86bd1c644fec011e2cb48414d66a7dbc3f7f1d

@brendanburns